src/darwin/Framework/CHIP/CHIPDeviceController.mm

```diff
@@ -748,11 +748,17 @@ - (CHIP_ERROR)isRunningOnFabric:(chip::FabricInfo *)fabric isRunning:(BOOL *)isR
         return CHIP_NO_ERROR;
     }
 
-    chip::Credentials::P256PublicKeySpan ourRootPublicKey, otherRootPublicKey;
-    ReturnErrorOnFailure(ourFabric->GetRootPubkey(ourRootPublicKey));
-    ReturnErrorOnFailure(fabric->GetRootPubkey(otherRootPublicKey));
+    chip::Credentials::FabricTable * fabricTable = _cppCommissioner->GetFabricTable();
+    if (!fabricTable) {
+        // Surprising as well!
+        return CHIP_ERROR_INCORRECT_STATE;
+    }
+
+    chip::Credentials::P256PublicKey ourRootPublicKey, otherRootPublicKey;
+    ReturnErrorOnFailure(fabricTable->FetchRootPubkey(ourFabric->GetFabricIndex(), ourRootPublicKey));
+    ReturnErrorOnFailure(fabricTable->FetchRootPubkey(fabric->GetFabricIndex(), otherRootPublicKey));
```

Comment:
This is not going to work right.  `fabric` comes from a different fabric table right now....

Reply:
Actually, it will, sort of, because the fabric tables use the same storage and have the same data.  I'll do a followup to make this saner, but this is OK for now.